qt/src/SourceManager.cc

```diff
@@ -142,48 +138,68 @@ int SourceManager::addSources(const QStringList& files) {
 	return added;
 }
 
-bool SourceManager::querySourcePassword(const QString& filename, QByteArray& password) const {
+bool SourceManager::querySourcePassword(const QString& filename, QByteArray& password, Poppler::Document* document) const {
 	bool success = true;
-	if(filename.endsWith(".pdf", Qt::CaseInsensitive)) {
-		std::unique_ptr<Poppler::Document> document(Poppler::Document::load(filename));
-		if(document && document->isLocked()) {
-			success = false;
-			bool ok = false;
-			QString message = QString(_("Enter password for file '%1':")).arg(QFileInfo(filename).fileName());
-			QString text;
-			while(true) {
-				text = QInputDialog::getText(MAIN, _("Protected PDF"), message, QLineEdit::Password, text, &ok);
-				if(!ok) {
-					break;
-				}
-				if(!document->unlock(text.toLocal8Bit(), text.toLocal8Bit())) {
-					password = text.toLocal8Bit();
-					success = true;
-					break;
-				}
+	if(document->isLocked()) {
+		success = false;
+		bool ok = false;
+		QString message = QString(_("Enter password for file '%1':")).arg(QFileInfo(filename).fileName());
+		QString text;
+		while(true) {
+			text = QInputDialog::getText(MAIN, _("Protected PDF"), message, QLineEdit::Password, text, &ok);
+			if(!ok) {
+				break;
+			}
+			if(!document->unlock(text.toLocal8Bit(), text.toLocal8Bit())) {
+				password = text.toLocal8Bit();
+				success = true;
```

Comment:
return true; and you can avoid the `success` variable

Reply:
Fixed.